lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
In addition, I don't think that all libraries require the threaded Fortran runtime, so not sure how one can have access to both w/out changing the compilers.yaml file.  With a new compiler definition, one can choose between the _r or non _r XL compilers with a compiler dependency.